### PR TITLE
docs: Regenerate Table of Contents in README  (issue no:  #929)

### PR DESCRIPTION
**Pull request by ap0calypse8**
Regenerated the Table of Contents (ToC) in the README to reflect the current structure. Used doctoc for automation . Verified the links are working correctly.

### CODE REVIEW

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
It should be below image and badges :)

**Comment by ap0calypse8** [comment]
i updated the TOC to be below image and badges

**Comment by ap0calypse8** [comment]
Please review the changes and let me know

File: README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
+
 <div align="center">
   <img src="https://raw.githubusercontent.com/cieslarmichal/faker-cxx/main/docs/public/logo.png" width="250" alt="C++ Faker, a powerful tool for generating realistic and randomized fake data in C++ programming, enhancing data testing and development workflows"/>
-  <h1>Faker C++</h1>
+  <h1>Faker CPP</h1>
```
Comment (cieslarmichal) [inline README.md:4]: Revert please

**Review by cieslarmichal — CHANGES_REQUESTED** [review]

**Review by cieslarmichal — APPROVED** [review]

**Pull request merged**